commons/src/main/java/com/powsybl/commons/json/JsonUtil.java

```diff
@@ -217,6 +217,48 @@ public static void writeOptionalIntegerField(JsonGenerator jsonGenerator, String
         return notFound;
     }
 
+    public static <T extends Extendable> List<Extension<T>> updateExtensions(JsonParser parser, DeserializationContext context,
+                                                                             ExtensionProviders<? extends ExtensionJsonSerializer> supplier, T toUpdatedParameters) throws IOException {
+        return updateExtensions(parser, context, SUPPLIER.get(), null, toUpdatedParameters);
+    }
+
+    public static <T extends Extendable> List<Extension<T>> updateExtensions(JsonParser parser, DeserializationContext context,
+                                                                             ExtensionProviders<? extends ExtensionJsonSerializer> supplier, Set<String> extensionsNotFound, T toUpdatedParameters) throws IOException {
+        Objects.requireNonNull(parser);
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(supplier);
+        Objects.requireNonNull(toUpdatedParameters);
+
+        List<Extension<T>> extensions = new ArrayList<>();
+        while (parser.nextToken() != JsonToken.END_OBJECT) {
+            Extension<T> extension = updateExtension(parser, context, supplier, extensionsNotFound, toUpdatedParameters);
+            if (extension != null) {
+                extensions.add(extension);
+            }
+        }
+        return extensions;
+    }
+
+    public static <T extends Extendable, E extends Extension<T>> E updateExtension(JsonParser parser, DeserializationContext context,
```

Comment:
the only method calling this one have already the requirements (non null) and there is no require non null on the method "readExtension" which is similar, maybe requirements can be add to both methods ?

Reply:
Or both methods can be private ?

Reply:
Yes, if this method is only called in `updateExtensions`, it might be better to keep it private. I think `readExtension` is public because it is called elsewhere (not in this repository but in another project using powsybl-core) but if `Objects.requireNonNull` is not called then, it is by mistake I think.